### DISCUSSION TITLE
OCP repo discussion  for RIWG

### DISCUSSION THREAD

**Discussion by AdrianHammond** [finos/compliant-financial-infrastructure#315] [General]
Releated to issue: https://github.com/finos/ansible-cfi-ocp/issues/4

Since split of code/doc out of the main [CFI repo](https://github.com/finos/compliant-financial-infrastructure/tree/main) into the [Ansible repo](https://github.com/finos/ansible-cfi-ocp) looks like branches have got out of sync and PR into Ansible repo failed. Below are notes from my investigation into potential issues. 

Outcome is for OCP SAA to be only asset in main CFI repo and for all OCP on GCP asests to be in the ansible-cfi-ansible repo - ref issue https://github.com/finos/ansible-cfi-ocp/issues/4

Current state:

- CFI (Main) last OCP update 7 months ago, missing update on OCP v4.11 and move of assets to ansible-cfi-ocp
- CFI (Main) 2 commits ahead / 63 commits behind 
- CFI (Dev) OCP doc/code removed, just contains OCP SSA (as expected) - https://github.com/finos/compliant-financial-infrastructure/tree/dev/accelerators/kubernetes/ocp
- CFI (Main) ahead #248 broken links in meeting agenda (Aug) and #278 merge dev to main (Nov) - https://github.com/finos/compliant-financial-infrastructure/compare/dev...main
- CFI (Main) 63 commits behind. Various updates including Johana update to OCP 4.11 - https://github.com/finos/compliant-financial-infrastructure/compare/main...dev
- ansible-cfi-ocp (main) as per orginal commit, no changes - https://github.com/finos/ansible-cfi-ocp
- ansible-cfi-ocp (dev) as per original commit, no changes - https://github.com/finos/ansible-cfi-ocp/tree/dev
- ansible-cfi-ocp (main and dev) has FINOS repo template applied so number of standard files, are these needed?

Potential steps to resolution as Johana has her fork with OCP 4.11 updates.

- [x] review with @thinkl33t @eddie-knight and agree following steps- working session schedule for 22/02/23
- [x] merge cfi dev into main, check they additional 2 commits are not regressed and check if they need to be applied to dev. At this point cfi should be at desired state of just having OCP SAA
- [ ] update ansible-cfi-ocp as currently based on standard FINOS template (https://github.com/finos/ansible-cfi-ocp/issues/7)
- [ ] fork ansible-cfi-ocp and update with OCP 4.11 assets (https://github.com/finos/ansible-cfi-ocp/issues/4)
- [ ] update ansible-cfi-ocp to OCP 4.12 (https://github.com/finos/ansible-cfi-ocp/issues/6)

I think I know how this happened, my error, when we  meet later please could we also discuss dev/main merge approach?

Thanks Adrian 

@thinkl33t @eddie-knight for our discuss today. + @abdullahgarcia I will add you to the invite,

**Comment by thinkl33t** [comment]
- [ ] remove `dev` entirely and set the default branch back to `main`

**Comment by thinkl33t** [comment]
- [ ] we need to remove the CI from this repo that spins up eks on commit

**Comment by AdrianHammond** [comment]
Good discussion with Eddie/Bob - Bob to open issue to remove dev from CFI. Closing this discussion

**Comment by abdullahgarcia** [comment]
Apologies @AdrianHammond , I missed this. Please let me know if there's anything to follow up on.
